desk/app/diary.hoon

```diff
@@ -465,33 +465,38 @@
     |=  [=flag:d =diary:d]
     ?.  =(p.action group.perm.diary)  ~
     `flag
-  ?+    q.q.action  cor
+  =/  diff  q.q.action
+  ?+  diff  cor
       [%fleet * %del ~]
     ~&  "%diary: revoke perms for {<affected>}"
     %+  roll  affected
     |=  [=flag:d co=_cor]
     ^+  cor
-    %+  roll  ~(tap in p.q.q.action)
+    %+  roll  ~(tap in p.diff)
     |=  [=ship ci=_cor]
     ^+  cor
     =/  di  (di-abed:di-core:ci flag)
     di-abet:(di-revoke:di ship)
   ::
-      [%fleet * %del-sects *]
-    ~&  "%diary recheck permissions for {<affected>}"
-    %+  roll  affected
-    |=  [=flag:d co=_cor]
-    =/  di  (di-abed:di-core:co flag)
-    di-abet:di-recheck:di
+    [%fleet * %add-sects *]    (recheck-perms affected ~)
+    [%fleet * %del-sects *]    (recheck-perms affected ~)
+    [%channel * %edit *]       (recheck-perms affected ~)
+    [%channel * %del-sects *]  (recheck-perms affected ~)
+    [%channel * %add-sects *]  (recheck-perms affected ~)
   ::
-      [%channel * %del-sects *]
-    ~&  "%diary recheck permissions for {<affected>}"
-    %+  roll  affected
-    |=  [=flag:d co=_cor]
-    =/  di  (di-abed:di-core:co flag)
-    di-abet:di-recheck:di
+      [%cabal * %del *]
+    =/  =sect:g  (slav %tas p.diff)
+    %+  recheck-perms  affected
+    `(~(gas in *(set sect:g)) ~[p.diff])
   ==
 ::
+++  recheck-perms
+  |=  [affected=(list flag:d) sects=(unit (set sect:g))]
```

Comment:
NB: would calling this with an empty set in the null case be clearer than using a unit or is there a semantic difference between the two?

Reply:
hmm yeah we probably could use an empty set here